sendgrid/sendgrid.py

```
@@ -73,6 +77,27 @@ def _get_default_headers(self):
     def reset_request_headers(self):
         self.client.request_headers = self._get_default_headers()
 
+    def make_request(self, request_obj, query_params=None, method="get", request_body=None):
+        """
+        makes request using the given request object
+        If the response status code is 429 it sleeps for 'rate limit reset time - current time' and then makes another request.
+        Number of retries is set to 5
+
+        :param request_obj: Request object
+        :param query_params: Query parameters
+        :param method: get / post / put / patch / delete
+        :param request_body: Can be used if method is not get
+        :return: response of given request object
+        """
+        for i in range(0, self.rate_limit_retry):
+            response = getattr(request_obj, method)(request_body=request_body, query_params=query_params)
+            if response.status_code == self.RATE_LIMIT_RESPONSE_CODE:
+                rate_limit_reset = response.headers['X-Ratelimit-Reset']
+                time_to_sleep = int(rate_limit_reset) - int(time.time())
+                time.sleep(time_to_sleep)
+            else:
+                return response

```

Should we throw an exception if the retry strategy fails?